contracts/ClaimsManagerWithKlerosArbitration.sol

```diff
@@ -225,21 +221,148 @@ contract ClaimsManagerWithKlerosArbitration is
             ].extraData;
     }
 
+    function getKlerosArbitrationParametersForClaimDisputePair(
+        uint256 claimIndex,
+        uint256 disputeId
+    ) public view returns (IArbitrator arbitrator, bytes memory extraData) {
+        KlerosArbitrationParameters
+            storage klerosArbitrationParameters = klerosArbitrationParametersHistory[
+                claimIndexToKlerosArbitrationParametersHistoryIndex[claimIndex]
+            ];
+        arbitrator = klerosArbitrationParameters.arbitrator;
+        require(
+            klerosArbitratorToDisputeIdToClaimIndex[address(arbitrator)][
+                disputeId
+            ] == claimIndex,
+            "Invalid claim-dispute pair"
+        );
+        extraData = klerosArbitrationParameters.extraData;
+    }
+
+    function arbitrationCost() external view returns (uint256) {
+        return klerosArbitrator().arbitrationCost(klerosArbitratorExtraData());
+    }
+
+    function appealCost(uint256 claimIndex, uint256 disputeId)
+        external
+        view
+        returns (uint256)
+    {
+        (
+            IArbitrator arbitrator,
+            bytes memory extraData
+        ) = getKlerosArbitrationParametersForClaimDisputePair(
+                claimIndex,
+                disputeId
+            );
+        return arbitrator.appealCost(disputeId, extraData);
+    }
+
+    function disputeStatus(uint256 claimIndex, uint256 disputeId)
+        external
+        view
+        returns (IArbitrator.DisputeStatus)
+    {
+        (
+            IArbitrator arbitrator,
+
+        ) = getKlerosArbitrationParametersForClaimDisputePair(
+                claimIndex,
+                disputeId
+            );
+        return arbitrator.disputeStatus(disputeId);
+    }
+
+    function currentRuling(uint256 claimIndex, uint256 disputeId)
+        external
+        view
+        returns (uint256)
+    {
+        (
+            IArbitrator arbitrator,
+
+        ) = getKlerosArbitrationParametersForClaimDisputePair(
+                claimIndex,
+                disputeId
+            );
+        return arbitrator.currentRuling(disputeId);
+    }
+
+    function getSubCourt(uint96 subCourtId)
+        external
+        view
+        returns (uint256[] memory children, uint256[4] memory timesPerPeriod)
+    {
+        return
+            IExtendedKlerosArbitrator(address(klerosArbitrator())).getSubCourt(
+                subCourtId
+            );
+    }
+
+    function courts(uint256 subCourtId)
+        external
+        view
+        returns (
+            uint96 parent,
+            bool hiddenVotes,
+            uint256 minStake,
+            uint256 alpha,
+            uint256 feeForJuror,
+            uint256 jurorsForCourtJump
+        )
+    {
+        return
+            IExtendedKlerosArbitrator(address(klerosArbitrator())).courts(
+                subCourtId
+            );
+    }
+
+    function disputes(uint256 disputeId)
+        external
+        view
+        returns (
```

Comment:
I was testing locally and ran into some problems. The `MockKlerosArbitrator` that we deploy is a `CentralizedArbitratorWithAppeal.sol` contract, and these convenience functions are expecting a `KlerosLiquid.sol` contract.

E.g. The `Dispute` stuct have different shapes between the two, and the courts aren't a thing in `CentralizedArbitratorWithAppeal.sol`

So I think you still need to update:
- https://github.com/api3dao/claims-manager/blob/main/contracts/mock/MockKlerosArbitrator.sol
- and the deploy script https://github.com/api3dao/claims-manager/blob/main/test/ClaimsManagerWithKlerosArbitration.sol.js

to use a KlerosLiquid.sol contract?

Reply:
Yeah, I'm aware of this issue. I'll probably mock the additional functions that KlerosLiquid implements in MockKlerosArbitrator rather than using KlerosLiquid.

Reply:
Ah ok cool. FYI the `Dispute` structs have different shapes between the two contracts (which might complicate things):
```solidity
// CentralizedArbitratorWithAppeal.sol
    struct Dispute {
        IArbitrable arbitrated;
        uint256 choices;
        uint256 ruling;
        DisputeStatus status;
        uint256 appealPeriodStart;
        uint256 appealPeriodEnd;
        uint256 appealCount;
    }
```
```solidity
// KlerosLiquid.sol
    struct Dispute { // Note that appeal `0` is equivalent to the first round of the dispute.
        uint96 subcourtID; // The ID of the subcourt the dispute is in.
        Arbitrable arbitrated; // The arbitrated arbitrable contract.
        // The number of choices jurors have when voting. This does not include choice `0` which is reserved for "refuse to arbitrate"/"no ruling".
        uint numberOfChoices;
        Period period; // The current period of the dispute.
        uint lastPeriodChange; // The last time the period was changed.
        // The votes in the form `votes[appeal][voteID]`. On each round, a new list is pushed and packed with as many empty votes as there are draws. We use `dispute.votes.length` to get the number of appeals plus 1 for the first round.
        Vote[][] votes;
        VoteCounter[] voteCounters; // The vote counters in the form `voteCounters[appeal]`.
        uint[] tokensAtStakePerJuror; // The amount of tokens at stake for each juror in the form `tokensAtStakePerJuror[appeal]`.
        uint[] totalFeesForJurors; // The total juror fees paid in the form `totalFeesForJurors[appeal]`.
        uint drawsInRound; // A counter of draws made in the current round.
        uint commitsInRound; // A counter of commits made in the current round.
        uint[] votesInEachRound; // A counter of votes made in each round in the form `votesInEachRound[appeal]`.
        // A counter of vote reward repartitions made in each round in the form `repartitionsInEachRound[appeal]`.
        uint[] repartitionsInEachRound;
        uint[] penaltiesInEachRound; // The amount of tokens collected from penalties in each round in the form `penaltiesInEachRound[appeal]`.
        bool ruled; // True if the ruling has been executed, false otherwise.
    }
```

Reply:
Thanks for the heads up